parser/parser_test.go

```diff
@@ -1026,6 +1026,20 @@ func TestPosition(t *testing.T) {
 		is(parser.slice(node.Idx0(), node.Idx1()), "if (abc) { throw 'failed'; }")
 		node = node.(*ast.IfStatement).Consequent.(*ast.BlockStatement).List[0].(*ast.ThrowStatement)
 		is(node.Idx0(), 39)
+
+		parser = newParser("", "for (x=1; x<=4; x++) { console.log(x); }", 1, nil)
```

Comment:
We should warp the for so the `parser.slice` does meaningful work below.
```suggestion
		parser = newParser("", "(function(){ for (x=1; x<=4; x++) { console.log(x); } })", 1, nil)
```